src/components/Popover/index.tsx

```diff
@@ -117,17 +123,38 @@ export default function Popover({
   const updateCallback = useCallback(() => {
     update && update()
   }, [update])
-  useInterval(updateCallback, show ? 100 : null)
+  useInterval(updateCallback, show ? 200 : null)
+
+  useEffect(() => {
+    if (!delayBeforeShow) return
+    if (show) {
+      const tooltipTimer = setTimeout(() => {
+        setDelayShow(true)
+      }, delayBeforeShow)
+
+      return () => {
+        clearTimeout(tooltipTimer)
+      }
+    } else {
+      setDelayShow(false)
+      return
+    }
+  }, [show, delayBeforeShow])
 
   return showInline ? (
-    <PopoverContainer show={show}>{content}</PopoverContainer>
+    <PopoverContainer show={show && delayShow}>{content}</PopoverContainer>
```

Comment:
Can we incorporate `show` and `delayShow` into a single variable, something like `shouldShow`? It feels to me that `shouldShow` is always `true` when `show` is true, isn't it? I would think about that and try to simplify.

Reply:
One of the ways one could think about it is creating a higher-order-compoent that takes `show` prop, adds timeout and passes `show` prop down to the final component (like this Popover), so Popover itself is not aware of the "timeout" and "interval". 

At the same time, I understand our codebase is far from being well organised as far as components, so it may seem like an overkill. Just a general comment.

Reply:
Okayyy actually upon second look, we do not need to use the variable `show` anymore in line 147 and line 154, we can just use `shouldShow`. I changed it so the useEffect now massages all information into making shouldShow the single source of truth for when the popover should appear and disappear. 

IMO this makes it simple/elegant enough to merge in 🤔 @zzmp @tinaszheng 

see comment here: https://github.com/Uniswap/interface/pull/6240/files#r1159048428

Reply:
nvm removed this delay altogether